lib/remote_page.py

```diff
@@ -312,14 +312,18 @@ def _fix_relative_urls(self, url, tag_name, attr_name):
 
                 # url points to the exercise service, e.g., MOOC-Grader.
                 # This fixes links to static files (such as images) in RST chapters.
-                # The image URL must be absolute and refer to the grader server
-                # instead of the A+ server. A relative URL with only path
-                # "/static/course/image.png" would target the A+ server when
-                # it is included in the A+ page. The value should be a relative
+                # The image URL must be absolute and refer to the server with static content
+                # instead of the A+ server. Traditionally this has been the grader server, but
+                # recently gitmanager is used. If GITMANAGER_URL is specified, we assume gitmanager.
+                # A relative URL with only path "/static/course/image.png" would target the A+ server
+                # when it is included in the A+ page. The value should be a relative
                 # path in the course build directory so that it becomes the full
                 # correct URL to the target file.
                 # E.g., urljoin('http://localhost:8080/static/default/module1/chapter.html', "../_images/image.png")
                 # -> 'http://localhost:8080/static/default/_images/image.png'
+                if settings.GITMANAGER_URL:
+                    gmurl = urlparse(settings.GITMANAGER_URL)
+                    url = url._replace(scheme=gmurl.scheme, netloc=gmurl.netloc)
```

Comment:
What do you think of these?

* This `url` variable is re-assigned inside the for loop. You lose the original method parameter value. If `url` were used elsewhere inside the loop, you would need to be careful about changing the value here so that it does not affect the next loop iteration in unwanted ways.
* This repeats the same `url` modification on multiple loop iterations. Only one time could be enough.
* In local testing, `GITMANAGER_URL` is `http://gitmanager:8070`. However, that address is not intended for the web browser as the browser should use the address `http://localhost:8070`. This always uses the incorrect address that does not work in the web browser unless you set a mapping `127.0.0.1 gitmanager` in `/etc/hosts` (requires root privileges in the host machine).
  * A+ already has the workaround: setting `REMOTE_PAGE_HOSTS_MAP`. However, this new code ignores that setting and overrides the `url` netloc with `GITMANAGER_URL`.
  * https://github.com/apluslms/a-plus/blob/9c7e6073b9554e959140c6f125ee2faedc9c7e60/lib/remote_page.py#L133-L136
  * https://github.com/apluslms/run-aplus-front/blob/b14596727258a221c039f178df80670edac1dcc7/rootfs/srv/aplus-cont-settings.py#L102-L104


```
REMOTE_PAGE_HOSTS_MAP = {
    "grader:8080": "localhost:8080",
    "gitmanager:8070": "localhost:8070",
}
```